staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go

```diff
@@ -1039,37 +1039,58 @@ func (r *crdHandler) getOrCreateServingInfoFor(uid types.UID, name string) (*crd
 
 		scaleScopes[v.Name] = &scaleScope
 
-		// override status subresource values
-		// shallow copy
+		if v.Deprecated {
+			deprecated[v.Name] = true
+			if v.DeprecationWarning != nil {
+				warnings[v.Name] = append(warnings[v.Name], *v.DeprecationWarning)
+			} else {
+				warnings[v.Name] = append(warnings[v.Name], defaultDeprecationWarning(v.Name, crd.Spec))
+			}
+		}
+	}
+
+	// override status subresource values
+	// shallow copy
```

Comment:
This comment doesn't belong here, does it?

Reply:
This part of the code indeed modifies the status subresource. I moved the original code here, and therefore also transferred the corresponding comments.
Do you mean that I should move this comment to line 1054 or somewhere after it?